future-apps/actor/contracts/Actor.sol

```diff
@@ -0,0 +1,184 @@
+/*
+ * SPDX-License-Identitifer:    GPL-3.0-or-later
+ */
+
+pragma solidity 0.4.24;
+
+import "./SignatureValidator.sol";
+import "./standards/IERC165.sol";
+import "./standards/IERC1271.sol";
+
+import "@aragon/apps-vault/contracts/Vault.sol";
+
+import "@aragon/os/contracts/common/IForwarder.sol";
+
+
+contract Actor is Vault, IERC165, IERC1271, IForwarder {
+    bytes32 public constant EXECUTE_ROLE = keccak256("EXECUTE_ROLE");
+    bytes32 public constant RUN_SCRIPT_ROLE = keccak256("RUN_SCRIPT_ROLE");
+    bytes32 public constant PRESIGN_HASH_ROLE = keccak256("PRESIGN_HASH_ROLE");
+    bytes32 public constant DESIGNATE_SIGNER_ROLE = keccak256("DESIGNATE_SIGNER_ROLE");
+
+    bytes4 public constant ISVALIDSIG_INTERFACE_ID = 0xabababab; // TODO: Add actual interfaceId
+
+    string private constant ERROR_EXECUTE_ETH_NO_DATA = "VAULT_EXECUTE_ETH_NO_DATA";
+    string private constant ERROR_EXECUTE_TARGET_NOT_CONTRACT = "VAULT_EXECUTE_TARGET_NOT_CONTRACT";
+
+    uint256 internal constant STATICCALL_MAX_GAS = 50000;
+
+    mapping (bytes32 => bool) public isPresigned;
+    address public designatedSigner;
+
+    event Execute(address indexed sender, address indexed target, uint256 ethValue, bytes data);
+    event PresignHash(address indexed sender, bytes32 indexed hash);
+    event SetDesignatedSigner(address indexed sender, address indexed oldSigner, address indexed newSigner);
+
+    // TODO: requires the @decodeData helper to be implemented in radspec
+    /**
+    * @notice Execute `@decodeData(target, data)` on `target` `ethValue == 0 ? '' : '(Sending' + @formatToken(ethValue, ETH) + ')'`.
+    * @param _target Address where the action is being executed
+    * @param _ethValue Amount of ETH from the contract that is sent with the action
+    * @param _data Calldata for the action
+    * @return Exits call frame forwarding the return data of the executed call (either error or success data)
+    */
+    function execute(address _target, uint256 _ethValue, bytes _data)
+        external // This function MUST always be external as the function performs a low level return, exiting the Actor app execution context
+        authP(EXECUTE_ROLE, arr(_target, _ethValue, uint256(getSig(_data)))) // TODO: Test that sig bytes are the least significant bytes
+    {
+        require(_ethValue == 0 || _data.length > 0, ERROR_EXECUTE_ETH_NO_DATA); // if ETH value is sent, there must be data
+        require(isContract(_target), ERROR_EXECUTE_TARGET_NOT_CONTRACT);
```

Comment:
Is there a reason we don't want the actor app to be able to send eth to someone without calling a function? Anyone able to call the execute function will be able to send funds anyway, and this makes compatibility with certain dapps slightly more difficult.

Reply:
Thanks to Actor inheriting from Vault, that can be done using the `transfer` function. I just tried to separate role concerns here.